File: alt-anns/typewiz/plain.ts
function vertices(t) {
  if (t.op === "node") {
    return 1 + vertices(t.left) + vertices(t.right);
  } else if (t.op === "leaf") {
    return 1;
  } else {
    throw t.op;
  }
}

console.log(vertices({op: "node",
                      left: {op: "leaf", val: 1},
                      right: {op: "leaf", val: 2}}));

File: alt-anns/pytype/plain.py
def nodes(t):
    if t['op'] == 'node':
        return nodes(t['left']) + nodes(t['right']) + 1
    elif t['op'] == 'leaf':
        return 1
    else:
        raise Exception("no")

print(nodes({'op': 'node',
             'left': {'op': 'leaf', 'val': 1},
             'right': {'op': 'leaf', 'val': 2}}))
